code/model/DNProject.php

```diff
@@ -358,9 +363,19 @@ public function getRepository() {
 	 * @return ArrayList
 	 */
 	public function DNEnvironmentList() {
+
+		if(!self::$_current_member_cache) {
+			self::$_current_member_cache = Member::currentUser();
+		}
+
+		if(self::$_current_member_cache === false) {
+			new ArrayList();
```

Comment:
The code here is a little confusing. Couldn't you combine the `!` and `false` check into one?